### PR TITLE
[Publisher][Design Updates] Update Records overview page layout

### LINKED ISSUE #1353: [Publisher][Design Updates] Update Records overview page layout
Update the Records overview page UI.

Design: [Figma](https://www.figma.com/design/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?node-id=10069-2758&t=uLy05Ib0gxARkIz2-4)

Relevant Files (there may be more, these are starting points):
* Records overview page: [Reports.tsx](https://github.com/Recidiviz/justice-counts/blob/main/publisher/src/pages/Reports.tsx), [Reports.styles.tsx](https://github.com/Recidiviz/justice-counts/blob/main/publisher/src/components/Reports/Reports.styles.tsx)

Only this section here:

Please reach out to @mxosman if anything in the Figma is unclear (including but not limited to expected behavior), or anything at all regarding this project.

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

Updated Records overview page layout

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

closes #1353 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by mxosman** [comment]
@nasaownsky - this is looking SO GOOD! Wow!

Design notes:
1. Can we reduce the side padding in the bulk actions dropdown?
    - Screenshot 2024-07-17 at 11 26 22 AM

2. There's extra top padding on the table header - can we reduce that?
    - Screenshot 2024-07-17 at 11 27 09 AM [Figma]
    - Screenshot 2024-07-17 at 11 27 05 AM [This PR]

3. The header colors are lighter than the Figma - can we darken to match the Figma to help it be more readable?
4. Can we update the below modal as well?
    - Screenshot 2024-07-17 at 11 25 14 AM
    - Screenshot 2024-07-17 at 11 31 44 AM

File: publisher/src/utils/dateUtils.ts [hunk 1]
```diff
@@ -46,6 +46,20 @@ export const printDateAsMonthYear = (month: number, year: number): string => {
...
+/**
+ * @returns the month, day and year as a string
+ * @example "March 22, 2022"
+ */
+export const printDateAsDayMonthYear = (dateString: string): string => {
+  const date = new Date(dateString);
+
+  const day = date.getDate();
+  const month = monthsByName[date.getMonth()];
+  const year = date.getFullYear();
```
Comment (mxosman) [inline publisher/src/utils/dateUtils.ts:58]: Prefer UTC for all of these

**Comment by nasaownsky** [comment]
@mxosman, fixes arrived! I actually can't delete records, so could you please check if the changes to delete's modal is all right? Also updated `UnauthorizedDeleteActionModal` so it matches the rest of the modals here.

**Review by mxosman — APPROVED** [review]
@nasaownsky - it looks so damn good! Thank you so much for all the fixes - AND for updating the unauthorized modal - totally forgot that one! The delete modals look fantastic - I think the deleting action requires you to be a Justice Counts Admin (which you can set via the admin panel for an agency if you wanted to).

I appreciate your help with this so much!! 🙏

**Pull request merged**